File: JavaScript/0001-two-sum.js
function twoSum (nums, target) {
    const valueToIndex = new Map();
    for (let i=0; i < nums.length; i++){
        const complement = target - nums[i];
        if (valueToIndex.has(complement)) {
            return [valueToIndex.get(complement),i];
        }
        valueToIndex.set(nums[i],i);
    }
    return [];
}

console.log(twoSum([2, 7, 11, 15], 9)); 

File: Java/0001-two-sum.java
import java.util.HashMap;
import java.util.Map;

public class Solution {

    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> valueToIndex = new HashMap<>();
        for (int i=0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (valueToIndex.containsKey(complement)){
                return new int[] { valueToIndex.get(complement),i};
            }
            valueToIndex.put(nums[i],i);
        }
        return new int[0]
    }
}